### PR TITLE
Compute InitLaraFirestoreParams from context.

### PR DESCRIPTION
**Pull request by knowuh**
This work is not yet complete, but it does the job of pulling in the required configuration information from the portal and from LARA and initializing Firebase  before rendering the react component. I have questions about how to correctly initialize firebase. TBD.

In order to get this to "work" locally (where work just means printing the store configuration in the console) you will need to:

* add a firebaseApp secret to your portal.  
* Add the plugin to an activity.
* Configure the plugin by specifying the same firebaseAppName in JSON for the authoring data of the plugin, eg:
`{ "firebaseAppName" : "plugin-test" }`

This is part of:

[#161772107]  Display Publish/Sharing Option for Students

https://www.pivotaltracker.com/story/show/161772107

### CODE REVIEW

File: src/lara/helper-functions.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    const pluginId = context.pluginId;
+    const portalUserId = portalUserPathToFirebaseId(portalClaims.user_id);
+    const userMap: SharedClassUserMap = {};
+    const interactiveName = interactiveState.interactive_name;
+    const classHash = classInfo.class_hash;
+    // TODO: I don't think we need to initialize this from here....
+    // This would erase all share info
+    // classInfo.students.forEach( (student) => {
+    //   const key = portalUserPathToFirebaseId(student.id);
+    //   userMap[key] = undefined;
```
Comment (dougmartin) [inline src/lara/helper-functions.ts]: This is meant to map a userId to a displayName so you can replace the `undefined` with something like \`${student.first_name} ${student.last_name[0].toUpperCase()}.\`

File: src/plugin.tsx [hunk 2]
```diff
@@ -27,12 +36,24 @@ const getAuthoredState = (context: IExternalScriptContext) => {
...
+    Promise.all([
+      getFirebaseJWT(context, this.authoredState.firebaseAppName || DefaultFirebaseApp),
+      getClassInfo(context),
+      getInteractiveState(context)
+    ])
+    .then( ([jwtResponse, classInfo, interactiveState]) => {
+      const config = getFireStoreParams(context, jwtResponse, classInfo, interactiveState);
+      console.log(config);
+      store.init(config);
+      this.renderPluginApp();
```
Comment (dougmartin) [inline src/plugin.tsx]: You should be leave the `this.renderPluginApp()` invocation where you had it, outside the Promise.then callback

Comment (knowuh) [reply src/plugin.tsx]: Hmn.  Weird, when I tried that I got:

``` 
plugins.js?body=1:17 Error: Firestore store not initialized!
    at FirestoreStore../src/stores/firestore.ts.FirestoreStore.ensureInitalized (firestore.ts:164)
```

**Comment by knowuh** [comment]
@dougmartin -- Some changes in 31e6643

Couldn't mount the component without a store that had been initialized though.  I might be misunderstanding your intention.

**Pull request merged**